presto-main/src/test/java/io/prestosql/execution/TestCallTask.java

```diff
@@ -0,0 +1,138 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.execution;
+
+import com.google.common.collect.ImmutableList;
+import io.prestosql.Session;
+import io.prestosql.connector.CatalogName;
+import io.prestosql.execution.warnings.WarningCollector;
+import io.prestosql.metadata.CatalogManager;
+import io.prestosql.metadata.MetadataManager;
+import io.prestosql.security.AccessControl;
+import io.prestosql.security.AllowAllAccessControl;
+import io.prestosql.spi.procedure.Procedure;
+import io.prestosql.spi.resourcegroups.ResourceGroupId;
+import io.prestosql.spi.type.BigintType;
+import io.prestosql.sql.analyzer.FeaturesConfig;
+import io.prestosql.sql.tree.Call;
+import io.prestosql.sql.tree.CallArgument;
+import io.prestosql.sql.tree.LongLiteral;
+import io.prestosql.sql.tree.QualifiedName;
+import io.prestosql.transaction.TransactionManager;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.net.URI;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+
+import static io.airlift.concurrent.Threads.daemonThreadsNamed;
+import static io.prestosql.metadata.MetadataManager.createTestMetadataManager;
+import static io.prestosql.spi.block.MethodHandleUtil.methodHandle;
+import static io.prestosql.testing.TestingSession.createBogusTestingCatalog;
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static io.prestosql.transaction.InMemoryTransactionManager.createTestTransactionManager;
+import static java.util.concurrent.Executors.newCachedThreadPool;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestCallTask
+{
+    private final ExecutorService executor = newCachedThreadPool(daemonThreadsNamed("call-task-test-%s"));
+
+    private static Set<Long> invocations = new HashSet<>();
+
+    @AfterClass
+    public void close()
+    {
+        executor.shutdown();
+    }
+
+    @Test
+    public void testExecute()
+    {
+        TransactionManager transactionManager = createTransactionManager();
+        MetadataManager metadata = createMetadataManager(transactionManager);
+
+        AccessControl accessControl = new AllowAllAccessControl();
+        QueryStateMachine stateMachine = stateMachine(transactionManager, metadata, accessControl);
+
+        Call procedure = getProcedureInvocation(1);
+
+        new CallTask().execute(procedure, transactionManager, metadata, accessControl, stateMachine, ImmutableList.of());
+
+        assertThat(invocations).contains(1L);
```

Comment:
Maybe test with more meaningful (aka "more random") number. `1` is more likely to come from a different place than eg `2382344352`.

Reply:
I am not entirely sure about that. Just how many times when prompted for file name you typed "fadsf" and it turned out that the file already exists. Ideally we could make a static global atomic counter for those numbers but that seems to be an overkill.

Reply:
> Ideally we could make a static global atomic counter for those numbers but that seems to be an overkill.

i agree

> I am not entirely sure about that. Just how many times when prompted for file name you typed "fadsf" and it turned out that the file already exists.

Yes, but still better than "a.out"